internal/pkg/db/mongo/models/device.go

```diff
@@ -64,12 +67,19 @@ func (d *Device) ToContract(dsTransform deviceServiceTransform, dpTransform devi
 	result.AdminState = d.AdminState
 	result.OperatingState = d.OperatingState
 
-	aModel, err := aTransform.DBRefToAddressable(d.Addressable)
+	p := make(map[string]map[string]string)
```

Comment:
I notice a lot of single character variable names. Why not just use protocols in this instance? Makes it clear what it is going to be used for.

Reply:
In Go, similar to C, it's an idiom to be concise and so, similar to C, if the scope of a variable is short enough from declaration to usage then the name can be less descriptive. In this case, the lifespan of variable `p` is 6 lines. Also, `p` doesn't really have much conceptual weight. It's allocated as a bucket for unmarshalling a byte array. If we were to instead do something of importance with `p`, then yes I'd agree it should be called `protocols`.

Reply:
Looking at the diff above, using something more descriptive than "p" makes it more readable to me. In general I agree with being concise, but I don't interpret this as meaning use single letter variables all over the place.

Reply:
@tsconn23 , that is the part of C that Go devs have inherited that I dislike... ;-) Moving on... :-)